### ISSUE TITLE
Failedjson: cannot unmarshal object into Go value of type tasks.JiraUserApiRes

### ISSUE THREAD

**Issue by bdimitrovITGix** [merico-dev/lake#1039]
Hello,

I am unable to collect data from the connected Jira server. When running the trigger, with the proper "boardId" and "sourceId", the following error occurs:
`Failedjson: cannot unmarshal object into Go value of type tasks.JiraUserApiRes`

DevLake is running on an AWS EC2 instance with docker-compose. I have not made any custom changes and pretty much followed the How-To in the repository.
What additional information can I provide for debugging ?

**Comment by klesh** [comment]
Hello @bdimitrovITGix , sorry for your inconvenience.
Were you connect to the Jira Cloud or Jira Server?

**Comment by bdimitrovITGix** [comment]
> Hello @bdimitrovITGix , sorry for your inconvenience. Were you connect to the Jira Cloud or Jira Server?

Jira Cloud. Created a new user, gave him API TOKEN and access to the board. No settings.

**Comment by klesh** [comment]
Possibly caused by some missing fields on your side.

Can you query this api `https://merico.atlassian.net/rest/api/3/users/search` with your token using something like postman, and post the response here? Or screenshot is fine.
Remember to filter out the sensitive information if any.

We need only the structure to investigate the cause, like what keys did it actually return with your setup.

**Comment by bdimitrovITGix** [comment]
>

**Comment by klesh** [comment]
My guess is that the token you used has no permission to access users list.
No sure why, but, devlake should provide some kind of helpful message in this case nontherless.
@Startrekzky Can you take a look at this issue?

**Comment by Startrekzky** [comment]
> Hello, I am not really familiar with making API calls or Postman but I did some testing with simple cURL and Jira documentation. What I noticed is quite strange. First I tried a simple query to the API with CURL: `curl -D- \ -X GET \ -H "Authorization: Basic BASIC_TOKEN" \ -H "Content-Type: application/json" \ "https://merico.atlassian.net/rest/api/3/users/search" ` I also tried this against our OWN Jira API. Both resulted in the same error:
> 
> ```
> {"errorMessages":["error.no-permission"],"errors":{}}% 
> ```
> 
> After that I tried to query a single user my name matching:
> 
> ```
> curl -D- \
>    -X GET \
>    -H "Authorization: Basic BASIC_TOKEN" \
>    -H "Content-Type: application/json" \
>    "https://MY_DOMAIN.atlassian.net/rest/api/3/user/search?query=devlake"
> ```
> 
> The above query should match all users with Display Name and/or Email that equal "devlake". The return result is an empty string- "[]". This is per Jira API documentation. Lastly, I tried two more queries to verify that my user and token are working. When I queried a user my his ID Key it WORKED. I was also Able to query an ISSUE by it's KEY. It return the proper information.
> 
> I think that something has changed with Jira and it's user queries ? Maybe restrictions related to GDPR or something. It seems that I am not able to simply get all users.
> 
> P.S. I closed the issue by mistake.

Hi. Your second API call is successful. It returned an empty string- "[]" because you passed a parameter. Try the code snippet below and you should get all users.
```
curl -D- \
   -X GET \
   -H "Authorization: Basic BASIC_TOKEN" \
   -H "Content-Type: application/json" \
   "https://MY_DOMAIN.atlassian.net/rest/api/3/user/search
```

Well. It seems the API connection was successful. Can you please put up a screenshot of the JSON on the `Trigger Collection` page so we can check if this JSON file is properly configured?

**Comment by klesh** [comment]
@bdimitrovITGix Sorry, I forgot to tell you the `merico` should be replaced with you actual domain, can you follow what @Startrekzky said, and paste the output here, so we can observe what was wrong the struct. (remember to remove any sensitive information, we need only the sturcture to pinpoint the problem).
Sorry for your inconvenience.

**Comment by bdimitrovITGix** [comment]
Hello, 
It turns out that the Jira user was at fault. As @klesh pointed out it could not access the user list. I am not sure why exactly. I did the same thing and created a 2nd user. This time data collection was possible. Apologies for dragging everyone in this.
Still I would suggest a clearer error or maybe an option to see the API response of the collection.
Again, thanks to everyone.

**Comment by klesh** [comment]
related to #865

**Comment by klesh** [comment]
closed by #1098
